python_modules/dagster/dagster_tests/execution_tests/dynamic_tests/test_reexecute.py

```diff
@@ -139,3 +146,48 @@ def test_reexec_from_parent_3():
         assert reexec_result.result_for_solid("multiply_by_two").output_value() == {
             "2": 40,
         }
+
+
+def dynamic_job():
+    @op
+    def echo(x):
+        return x
+
+    @op
+    def adder(ls: List[int]) -> int:
+        return sum(ls)
+
+    @op(out=Out(is_required=False))
+    def add_one(context, i: int) -> int:
+        if (
+            context.pipeline_run.parent_run_id
+            and i % 2 == 0  # re-execution run skipped odd numbers
+            or not context.pipeline_run.parent_run_id
+            and i % 2 == 1  # root run skipped even numbers
+        ):
+            yield Output(i + 1)
+
+    @op(out=DynamicOut())
+    def dynamic_op():
+        for i in range(10):
+            yield DynamicOutput(value=i, mapping_key=str(i))
+
+    @job
+    def _dynamic_job():
+        dynamic_results = dynamic_op().map(lambda n: echo(add_one(n)))
+        adder(dynamic_results.collect())
+
+    return _dynamic_job
+
+
+def test_reexec_all_dynamic_with_partial_skip_fan_in():
```

Comment:
Did you verify that the test fails without the change?

Reply:
yes, it fails without the change